spec/controllers/api/orders_controller_spec.rb

```diff
@@ -156,6 +158,109 @@ module Api
+
+      context "Resource not found" do
+        before { allow(controller).to receive(:spree_current_user) { admin_user } }
+
+        it "when no order number is given" do
+          get :show, id: nil
+          expect_resource_not_found
+        end
+
+        it "when order number given is not in the systen" do
+          get :show, id: "X1321313232"
+          expect_resource_not_found
+        end
+
+        def expect_resource_not_found
+          expect(json_response).to eq( "error" => "The resource you were looking for could not be found." )
+          expect(response.status).to eq(404)
+        end
+      end
+
+      context "access" do
+        it "returns unauthorized, as a regular user" do
+          allow(controller).to receive(:spree_current_user) { regular_user }
+          get :show, id: order.number
+          assert_unauthorized!
+        end
+
+        it "returns the order, as an admin user" do
+          allow(controller).to receive(:spree_current_user) { admin_user }
+          get :show, id: order.number
+          expect_order
+        end
+
+        it "returns the order, as the order distributor owner" do
+          allow(controller).to receive(:spree_current_user) { order.distributor.owner }
+          get :show, id: order.number
+          expect_order
+        end
+
+        it "returns unauthorized, as the order product's supplier owner" do
+          allow(controller).to receive(:spree_current_user) { order.line_items.first.variant.product.supplier.owner }
+          get :show, id: order.number
+          assert_unauthorized!
+        end
+
+        it "returns the order, as the Order Cycle coorinator owner" do
+          allow(controller).to receive(:spree_current_user) { order.order_cycle.coordinator.owner }
+          get :show, id: order.number
+          expect_order
+        end
+      end
+
+      context "as distributor owner" do
+        let!(:order) { create(:completed_order_with_fees, order_cycle: order_cycle, distributor: distributor ) }
+
+        before { allow(controller).to receive(:spree_current_user) { order.distributor.owner } }
+
+        it "can view an order not in a standard state" do
+          order.update_attributes(completed_at: nil, state: 'shipped')
+          get :show, id: order.number
+          expect_order
+        end
+
+        it "returns an order with all required fields" do
+          get :show, id: order.number
+          expect_order
+          expect_detailed_attributes_to_be_present(json_response)
+
+          expect(json_response[:bill_address][:address1]).to eq(order.bill_address.address1)
+          expect(json_response[:bill_address][:lastname]).to eq(order.bill_address.lastname)
+          expect(json_response[:ship_address][:address1]).to eq(order.ship_address.address1)
+          expect(json_response[:ship_address][:lastname]).to eq(order.ship_address.lastname)
```

Comment:
you can do `expect(json_response[:bill_address]).to eq(order.bill_address.attributes)` instead.

Reply:
not here because the serializer is not using all the attributes, just a few of them.